### PR TITLE
ROX-18116: compliance check results proto stores

### PR DESCRIPTION
**Pull request by dashrews78**
## Description

Stores and protos for storing compliance scans and check results from the compliance operator

## Checklist
- [ ] Investigated and inspected CI test results
- [x] Unit test and regression tests added
~- [ ] Evaluated and added CHANGELOG entry if required~
~- [ ] Determined and documented upgrade steps~
~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs]~(https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

CI as just generated code.

### CODE REVIEW

File: pkg/search/options.go [hunk 1]
```diff
@@ -199,7 +199,12 @@ var (
 	ControlID = newFieldLabel("Control ID")
 	Control   = newFieldLabel("Control")
 
-	ComplianceOperatorVersion = newFieldLabel("Compliance Operator Version")
+	ComplianceOperatorVersion     = newFieldLabel("Compliance Operator Version")
+	ComplianceOperatorScanName    = newFieldLabel("Compliance Operator Scan Name")
```
Comment (md2119) [inline pkg/search/options.go:203]: I think `Compliance Scan Name` should be okay.

File: pkg/search/options.go [hunk 2]
```diff
@@ -199,7 +199,12 @@ var (
...
+	ComplianceOperatorSeverity    = newFieldLabel("Compliance Operator Severity")
```
Comment (md2119) [inline pkg/search/options.go:204]: `Compliance Rule Severity`

File: proto/storage/compliance_operator_v2.proto [hunk 3]
```diff
@@ -10,13 +10,36 @@ import "storage/user.proto";
...
+}
+
+// Represents the scan type whether a node or platform scan.
+enum ScanType {
+    UNSET_SCAN_TYPE = 0;
+    NODE_SCAN = 1;
+    PLATFORM_SCAN = 2;
+}
+
+message Profile {
```
Comment (md2119) [inline proto/storage/compliance_operator_v2.proto:27]: `ProfileSkin`, `ProfileShim`...(?)

File: proto/storage/compliance_operator_v2.proto [hunk 4]
```diff
@@ -10,13 +10,36 @@ import "storage/user.proto";
...
+    string                    profile_id               = 1 [(gogoproto.moretags) = 'search:"-" sql:"fk(ComplianceOperatorProfileV2:id),no-fk-constraint"'];
+}
+
+message ClusterScanStatus {
+    string                    cluster_id = 1 [(gogoproto.moretags) = 'search:"Cluster ID,hidden,store" sql:"fk(Cluster:id),no-fk-constraint,type(uuid)"'];
```
Comment (md2119) [inline proto/storage/compliance_operator_v2.proto:32]: Is this to be used to represent overall error or per profile or can be used for both?

File: proto/storage/compliance_operator_v2.proto [hunk 5]
```diff
@@ -46,33 +69,75 @@ message ComplianceOperatorRuleV2 {
...
+    string phase = 1;
+    string result = 2;
+    repeated string warnings = 3;
+}
+
+// Next Tag: 13
+// Scan object per cluster
+message ComplianceOperatorScanV2 {
+    string              id          = 1 [(gogoproto.moretags) = 'sql:"pk"'];
+    string              scan_name        = 2 [(gogoproto.moretags) = 'search:"Compliance Operator Scan Name,hidden,store" sql:"index=category:unique;name:scan_unique_indicator"'];
```
Comment (md2119) [inline proto/storage/compliance_operator_v2.proto:132]: fk? (this is bound to scan setting binding)

File: proto/storage/compliance_operator_v2.proto (hunk 4 above)
Comment (dashrews78) [reply proto/storage/compliance_operator_v2.proto:32]: The idea was per cluster.  For example we apply a schedule to 6 clusters but sensor has trouble applying the CRs with one.  We need to be able to see that.  It only applies to ScanSettings at the moment.  So I'm going to scope it to ScanSettings so it is more clear.  I wouldn't think we would need an error per profile on settings.

File: proto/storage/compliance_operator_v2.proto (hunk 5 above)
Comment (dashrews78) [reply proto/storage/compliance_operator_v2.proto:132]: good catch

File: proto/storage/compliance_operator_v2.proto [hunk 6]
```diff
@@ -46,33 +64,82 @@ message ComplianceOperatorRuleV2 {
...
+    string              check_name = 3;
+    string              cluster_id = 4 [(gogoproto.moretags) = 'search:"Cluster ID,hidden,store" sql:"fk(Cluster:id),no-fk-constraint,type(uuid)"'];
+    CheckStatus         status = 5  [(gogoproto.moretags) = 'search:"Compliance Check Status,hidden,store"'];
+    string              severity     = 6 [(gogoproto.moretags) = 'search:"Compliance Rule Severity,hidden,store"'];
+    string              description = 7;
+    string              instructions = 8;
+    map<string, string> labels = 9;
+    map<string, string> annotations = 10;
+    google.protobuf.Timestamp     created_time = 11;
+    string standard = 12 [(gogoproto.moretags) = 'search:"Standard,hidden,store"'];
```
Comment (md2119) [inline proto/storage/compliance_operator_v2.proto:119]: Is `Standard` currently used search label in compliance! lol!!

**Review by md2119 — APPROVED** [review]

**Pull request merged**